raster/r.kappa/calc_metrics.c

```diff
@@ -0,0 +1,213 @@
+#include <stdlib.h>
+#include <grass/gis.h>
+#include <grass/glocale.h>
+#include "kappa.h"
+#include "local_proto.h"
+
+static int longcomp(const void *aa, const void *bb);
+static int collapse(long *l, int n);
+
+
+void calc_metrics(void)
+{
+    int i, j, k;
+    long *clst;
+    int ncat1, ncat2;
+    int cndx;
+    double *pi, *pj, *pii;
+    double p0 = 0, pC = 0;
+    double inter1 = 0, inter2 = 0;
+    int a_i, b_i;
+
+    metrics = (METRICS *) G_malloc(sizeof(METRICS));
+    if (nstats == 0) {
+        G_warning(_("Both maps have nothing in common. Check the computational region."));
+        metrics->obs = 0;
+        metrics->correct = 0;
+        metrics->total_acc = 0;
+        metrics->kappa = -999;
+        metrics->kappa_var = -999;
```

Comment:
While NaN is not part of JSON, it is a common extension because it is part of JavaScript. Python [supports that by default](https://docs.python.org/3/library/json.html#json.dump) and db.univar and v.db.univar (#2386) behave that way.

Reply:
This is a tricky question. As it does not confirm with JSON spec, not all parsers are happy with NaN, NA, etc. -999 looks ugly but is compatible with current generation of R JSON parsers.
```
>>> json.dumps([1.0, 2.0, float('nan'), 3.0])
'[1.0, 2.0, NaN, 3.0]'
```
```
> fromJSON('[1.0, 2.0, NaN, 3.0]')
Error in fromJSON("[1.0, 2.0, NaN, 3.0]") : 
  not all data was parsed (9 chars were parsed out of a total of 20 chars)
```

Reply:
We could go with `null` although it is also not without downsides:
```
> fromJSON('[1.0, 2.0, -999.0, 3.0]')
[1]    1    2 -999    3
> fromJSON('[1.0, 2.0, null, 3.0]')
[[1]]
[1] 1

[[2]]
[1] 2

[[3]]
NULL

[[4]]
[1] 3
```